include/secp256k1.h

```diff
@@ -361,60 +419,113 @@ SECP256K1_WARN_UNUSED_RESULT int secp256k1_ec_seckey_verify(
  *           0: secret was invalid, try again
  */
 SECP256K1_WARN_UNUSED_RESULT int secp256k1_ec_pubkey_create(
-  const secp256k1_context_t* ctx,
-  secp256k1_pubkey_t *pubkey,
-  const unsigned char *seckey
+    const secp256k1_context_t* ctx,
+    secp256k1_pubkey_t *pubkey,
+    const unsigned char *seckey
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
 
-/** Export a private key in DER format.
- * In: ctx: pointer to a context object, initialized for signing (cannot be NULL)
+/** Export a private key in BER format.
+ *
+ *  In: ctx:         pointer to a context object, initialized for signing (cannot
+ *                   be NULL)
+ *      seckey:      pointer to a 32-byte secret key to export.
+ *      compressed:  whether the key should be exported in compressed format.
+ *  Out: privkey:    pointer to an array for storing the private key in BER.
+ *                   Should have space for 279 bytes, and cannot be NULL.
+ *       privkeylen: Pointer to an int where the length of the private key in
+ *                   privkey will be stored.
+ *  Returns: 1 if the private key was valid.
+ *
+ *  This function is purely meant for compatibility with applications that
+ *  require BER encoded keys. When working with secp256k1-specific code, the
+ *  simple 32-byte private keys are sufficient.
+ *
+ *  Note that this function does not guarantee correct DER output. It is
+ *  guaranteed to be parsable by secp256k1_ec_privkey_import.
  */
 SECP256K1_WARN_UNUSED_RESULT int secp256k1_ec_privkey_export(
-  const secp256k1_context_t* ctx,
-  const unsigned char *seckey,
-  unsigned char *privkey,
-  int *privkeylen,
-  int compressed
+    const secp256k1_context_t* ctx,
+    const unsigned char *seckey,
+    unsigned char *privkey,
+    int *privkeylen,
+    int compressed
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4);
 
-/** Import a private key in DER format. */
+/** Import a private key in DER format.
+ * Returns: 1 if a private key was extracted.
+ * In:  ctx:        pointer to a context object (cannot be NULL).
+ *      privkey:    pointer to a private key in DER format (cannot be NULL).
+ *      privkeylen: length of the DER private key pointed to be privkey.
+ * Out: seckey:     pointer to a 32-byte array for storing the private key.
+ *                  (cannot be NULL).
```

Comment:
Any reason this functions in/out parameters are not in the same order as the documentation?